admin/bin/check-ps-report-for-corrupt-data.js

```diff
@@ -0,0 +1,380 @@
+}
+
+function reportMissing (row, prop) {
+  process.stderr.write(`${prop} field missing for ${row['AttemptId']}\n`)
+}
+
+function reportProgress (count) {
+  // process.stdout.write('.')
+}
+
+function regexTest (row, prop, regex) {
+  if (row.hasOwnProperty(prop)) {
+    const val = row[prop]
+    if (val.length && !regex.test(val)) {
+      report(prop, row, val)
+    }
+  } else {
+    console.log(`Missing property: ${row['checkCode']} property: ${prop}`)
+  }
+}
+
+const checkID = function (row, prop) {
+  const regex = /^\d{1,2} x \d{1,2}$/
+  regexTest(row, prop, regex)
+}
+
+const checkResponse = function (row, prop) {
+  const regex = /^\d{0,5}$/
+  regexTest(row, prop, regex)
+}
+
+const checkK = function (row, prop) {
+  const val = row[prop]
+  const regex = /^(k|m|t|x)?\[(.*)\]*$/
+  const inputs = val.split(', ')
+  inputs.forEach(input => {
+    const s = input.trim()
+    if (s !== '' && !regex.test(s)) {
+      report(prop, row, val)
+    }
+  })
+}
+
+const checkSco = function (row, prop) {
+  const regex = /^(1|0)$/
+  regexTest(row, prop, regex)
+}
+
+function checkResponseTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOut (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutResponse (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutSco (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checktLoad (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktFirstKey (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktLastKey (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkOverallTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkRecallTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkReaderStart (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkReaderEnd (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checkDob (row, prop) {
+  const regex = /^\d\d\/\d\d\/\d\d\d\d$/
+  if (!row[prop]) {
+    return report(prop, row, row[prop])
+  }
+  regexTest(row, prop, regex)
+}
+
+function checkGender (row, prop) {
+  const val = row[prop]
+  if (val !== 'M' && val !== 'F') {
+    report(prop, row, val)
+  }
+}
+
+function checkPupilId (row, prop) {
+  const val = row[prop]
+  if (!row.hasOwnProperty(prop)) {
+    return reportMissing(row, prop)
+  }
+  if (val.length !== 13) {
+    report(prop, row, val)
+  }
+}
+
+function checkForename (row, prop) {
```

Comment:
can you just reuse some of these checks for some fields? i.e. checkForename and checkSurname

Reply:
Refactored and clarified.